src/backend/expungeservice/endpoints/users.py

```diff
@@ -2,24 +2,50 @@
 from flask import request, abort, jsonify
 from werkzeug.security import generate_password_hash
 
-# TODO temp hack - replace with table from DB
-users = {}
+from flask import g
+from expungeservice.database.user import create_user, get_user_by_email
+from expungeservice.endpoints.auth import auth_required
+from psycopg2.errors import UniqueViolation
+
 
 class Users(MethodView):
+    @auth_required
     def post(self):
         data = request.get_json()
 
         if data == None:
             abort(400)
 
-        # TODO temp hack - replace with table from DB
-        if data['email'] in users:
-            return 'User already exists! Please login', 202
+        """
+        - the @auth_required decorator identifies the logged in user. If no auth, return 401 UNAUTHORIZED.
+        - Verify the logged in user is admin. if not, return 403 FORBIDDEN.
+        - Verify the password meets the 8-character length requirement. If not, return 422
+        - Attempt to insert the user into the database. If failure due to duplicate entry, return 422
+        - If success, return the new user's email, admin flag, and creation timestamp.
+        """
+
+        authorized_user_db_result = get_user_by_email(g.database, g.auth_hdr_payload['sub'])
+        if not authorized_user_db_result['admin']:
```

Comment:
This will result in a 500 error if the user can't be found and `get_user_by_email` returns `None`, right? Seems at least worthwhile to comment, even if this is an edge case.

Reply:
The auth_required decorator checks that the named the user exists just before this database fetch is called. An error would throw if the user is deleted from the database between these two ops. I feel like this is safe enough to not worry about it.

Reply:
I agree we shouldn't do anything about it in the logic itself! But I'd say just take the reply you wrote ^ and paste it in to the code as a comment. Takes 10 seconds and you'll probably save people time in the future.

Reply:
@maxkwallace I didn't put this in because my refactor makes it irrelevant: the check for admin status is made on the same db result -- in the admin_auth_required function.

Reply:
Sounds good! Thanks for pointing that out!